client/my-sites/plans/jetpack-plans/plans-header.tsx

```diff
@@ -70,7 +70,9 @@ const PlansHeader = ( { context, shouldShowPlanRecommendation }: HeaderProps ) =
 	const purchasedProducts =
 		useSelector( ( state ) => getSiteProducts( state, siteId ) )
 			?.map( ( { productSlug } ) => productSlug )
-			.filter( ( productSlug ) => JETPACK_PRODUCTS_LIST.includes( productSlug ) ) ?? [];
+			.filter( ( productSlug ) =>
+				( JETPACK_PRODUCTS_LIST as ReadonlyArray< string > ).includes( productSlug )
+			) ?? [];
```

Comment:
another note here about fixing type errors _after_ a chain of computation or before. it looks like we're only concerned if `getSiteProducts(…)` returns things. although it would require adding curly-brackets, it's probably useful here to extract and abort early instead of carrying the nullability/optionality through the whole stack

```ts
useSelector( ( state ) => {
	const products = getSiteProducts( state, siteId );
	if ( ! products ) {
		return [];
	}

	return products.map( p => p.productSlug ).filter( slug => … );
} );
```

another note here too that could simplify the types would be to create a set of the products and check for set membership. it involves an additional up-front step but makes the filtering more semantic. probably best not for this kind of PR though

```ts
const JetpackProductSlugs = new Set( JETPACK_PRODUCTS_LIST as string[] );

…

const PlansHeader = … => {
	…
	const purchasedProducts = useSelector( ( state ) => getSiteProducts( state, siteId )
		.filter( p => JetpackPlanSlugs.has( p.productSlug ) )
		.map( p => p.productSlug );
	…
}
```